honeybadgermpc/hbavss.py

```diff
@@ -538,22 +541,71 @@ def _get_dealer_msg(self, values, n):
             # dispersal_msg_list: the list of payload z
             broadcast_msg, dispersal_msg_list = self._get_dealer_msg(values, n)
 
-        tag = f"{dealer_id}-{avss_id}-B-AVID"
+        tag = f"{dealer_id}-{avss_id}-B-RBC"
         send, recv = self.get_send(tag), self.subscribe_recv(tag)
 
         logger.debug("[%d] Starting reliable broadcast", self.my_id)
         rbc_msg = await reliablebroadcast(
             tag, self.my_id, n, self.t, dealer_id, broadcast_msg, recv, send)
 
+        tag = f"{dealer_id}-{avss_id}-B-AVID"
+        send, recv = self.get_send(tag), self.subscribe_recv(tag)
+
         logger.debug("[%d] Starting AVID disperse", self.my_id)
         avid = AVID(n, self.t, dealer_id, recv, send, n)
-        # start disperse in the background
-        self.avid_msg_queue.put_nowait((avid, tag, dispersal_msg_list))
 
         if client_mode and self.my_id == dealer_id:
             # In client_mode, the dealer is not supposed to do
             # anything after sending the initial value.
+            await avid.disperse(tag, self.my_id, dispersal_msg_list, client_mode=True)
+            self.shares_future.set_result(True)
             return
 
+        # start disperse in the background
+        self.avid_msg_queue.put_nowait((avid, tag, dispersal_msg_list))
+
         # avss processing
         await self._process_avss_msg(avss_id, dealer_id, rbc_msg, avid)
+
+
+def get_avss_params(n, t):
+    g, h = G1.rand(seed=[0, 0, 0, 1]), G1.rand(seed=[0, 0, 0, 1])
+    public_keys, private_keys = [None]*n, [None]*n
+    for i in range(n):
+        private_keys[i] = ZR.random(0)
+        public_keys[i] = pow(g, private_keys[i])
+    return g, h, public_keys, private_keys
+
+
+async def _run(peers, n, t, my_id):
+    g, h, pks, sks = get_avss_params(n + 1, t)
+    async with ProcessProgramRunner(peers, n+1, t, my_id) as runner:
+        send, recv = runner.get_send_recv('1')
+        crs = gen_pc_const_crs(t, g=g, h=h)
```

Comment:
move this till `592` out of the `with`. Basically start with `HbAVSS` directly in the `with` (no need to unnecessarily setup network connection if we don't want to use them right away.